jobclient/src/main/java/com/twosigma/cook/jobclient/Constraint.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Two Sigma Open Source, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.cook.jobclient;
+
+import com.google.common.base.Preconditions;
+import org.json.JSONArray;
+
+/**
+ * A simple class to represent a constraint.
+ *
+ * A constraint has 3 fields: an attribute, an operator, and a value.
+ * An example of constraints could be: "host","EQUALS", "foo.bar.com".
+ *
+ * For now, only "EQUALS" operator is supported by Cook. One could construct
+ * a constraint with an "EQUALS" operator by
+ *
+ * <pre>
+ *  Constraint c1 = Constraint.EQUALS.apply("host", "foo.bar.com");
+ *  Constraint c2 = new Constraint("host", Constraint.EQUALS, "foo.bar.org");
+ * </pre>
+ */
+final public class Constraint {
+
+    public enum Operator {
+        EQUALS("EQUALS");
+
+        Operator(String name) {
+        }
+
+        /**
+         * Returns a constraint by "applying" this operator to the given attribute and value.
+         *
+         * @param attribute The constraint attribute
+         * @param value     The constraint value
+         * @return a constraint.
+         */
+        public Constraint apply(String attribute, String value) {
```

Comment:
Do we expect all operators we add in the future to live up to this specification?

Reply:
That is good question but I guess I am not the right person to answer it :-)

Reply:
Since we are not sure, can we avoid this restriction then?

Reply:
You could imagine an operator "IN" which takes a list of values. @shamsimam, @WenboZhao  what api would you imagine in that case?

Reply:
I will refactor those ```Operator``` to e.g. 
```
OneToZeroOperator;   \\ E.g. UNIQUE
OneToOneOperator;    \\ E.g. EQUALS
OneToManyOperator;  \\ E.g. IN
```

Reply:
My vote would be to do away with the `apply` method and have an external factory method that builds the constraint for each specific operator as the Java generics support will not be able to help us here.

Reply:
hmm, actually, the `Constraint.XXX.apply` way is doing a factory pattern here. Although I admit that `build` instead `apply` is a better for java style naming convention. `apply()` is too scala-ish.

Reply:
I am not against factory pattern that the `apply`/`build` methods offer. My issue is with the signature of those methods. If we keep the operators as enums (I like that they are enums currently), we do not have a good way to deal with individual signatures for the different operators. So, we may need to introduce explicit factory methods in `Constraint` like `buildEqualsConstraint(String, String)` and `buildInConstraint(String, Collection<String>)`